### PR TITLE
add tfsec on PR

### PR DESCRIPTION
**Pull request by kjgarza**
enables static analysis tool to detect potential security risks.

`tfsec`, a static analysis tool that scans Terraform templates to find potential security issues.

## Purpose

closes: _Add github issue that originated this PR_

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)

- [x] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Reviewer, please remember our [guidelines](https://datacite.atlassian.net/wiki/spaces/TEC/pages/1168375809/Pull+Request+Guidelines):

- Be humble in the language and feedback you give, ask don't tell.
- Consider using positive language as opposed to neutral when offering feedback. This is to avoid the negative bias that can occur with neutral language appearing negative.
- Offer suggestions on how to improve code e.g. simplification or expanding clarity.
- Ensure you give reasons for the changes you are proposing.

### CODE REVIEW

File: .github/workflows/tfsec.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      contents: read
+      pull-requests: write
+
+    steps:
+      - name: Clone repo
+        uses: actions/checkout@master
+      - name: tfsec
+        uses: aquasecurity/tfsec-pr-commenter-action@v1.2.0
+        with:
+          github_token: ${{ github.token }}
```
Comment (digitaldogsbody) [inline .github/workflows/tfsec.yml:19]: I think this should be `github_token: ${{ secrets.GITHUB_TOKEN }}`

**Review by digitaldogsbody — CHANGES_REQUESTED** [review]

File: .github/workflows/tfsec.yml (hunk 1 above)
Comment (kjgarza) [reply .github/workflows/tfsec.yml:19]: thanks

**Comment by digitaldogsbody** [comment]
A couple of quick questions:
1) Have you run this against the existing code?
  a. If so, did it pick up any issues?
  b. If not, could we do it before merge to see what the current state is?
2) At the minute, it is configured to just run all tests - do we need this, or are there certain tests we should ignore?

And one slightly more major thought:
This workflow is configured to only run on pull requests (which makes sense because it works by commenting, similar to our Cypress integration), but the majority of changes to this repo are automatic, from other GitHub actions, and commit directly to the main branch. Has there been any consideration to a way of testing that?

**Comment by kjgarza** [comment]
https://github.com/aquasecurity/tfsec

```
cd mastino
tfsec .
```
I had run it. we have loads of security issues. That's mostly why I put it on PR-only ATM, then people can check before adding insecure code.

**Pull request merged**